### PR TITLE
Misc updates for Hive 0.11

### PR DESCRIPTION
**Pull request by harveyfeng**
- Detect LATERAL VIEW in ColumnPruner by looking for a LateralViewForwardOperator coupled with a TableScanOperator
- Remove stats\* unit tests from Hive - they're a continuous WIP

### CODE REVIEW

**Comment by harveyfeng** [comment]
Jenkins, retest this please!

**Comment by harveyfeng** [comment]
Alright, just need to re-patch windowing functions

File: src/test/tests_pass.txt [hunk 1]
```diff
@@ -766,3 +751,909 @@ testCliDriver_union_lateralview
 
 testCliDriver_union_view
 
+# New from Hive 0.11:
```
Comment (rxin) [inline src/test/tests_pass.txt]: Most of the following tests are not new though?

**Comment by ChengXiangLi** [comment]
looks good, i'm cool with this PR.

File: src/test/tests_pass.txt (hunk 1 above)
Comment (harveyfeng) [inline src/test/tests_pass.txt]: A *right outer join would be useful here

Comment (rxin) [inline src/test/tests_pass.txt]: note that you can do a sort, and then diff, and then leave only the diff (minus)

Comment (harveyfeng) [inline src/test/tests_pass.txt]: just noticed there's `.q` in some lines, so gotta clean this too

File: src/main/scala/shark/execution/optimization/ColumnPruner.scala [hunk 2]
```diff
@@ -124,8 +124,17 @@ class ColumnPruner(@transient op: TopOperator[_], @transient tbl: Table) extends
     }
 
     // recurse on the subtree
-    op.childOperators.foreach { y =>
-      computeColumnsToKeep(y, cols, op)
+    val numChildren = op.childOperators.size
+    var currentChildIndex = 0
+    while (currentChildIndex < numChildren) {
```
Comment (rxin) [inline src/main/scala/shark/execution/optimization/ColumnPruner.scala:129]: Can you add a comment to say what this is doing (i.e. lateral view is a special case blah blah blah).

**Comment by rxin** [comment]
Feel free to merge once you addressed the previous comment.

**Comment by rxin** [comment]
Thanks!

**Pull request merged**